pkg/csi/cinder/controllerserver.go

```diff
@@ -55,6 +55,12 @@ func (cs *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 		return nil, status.Error(codes.InvalidArgument, "[CreateVolume] missing Volume capability")
 	}
 
+	for _, cap := range volCapabilities {
```

Comment:
Feels like some refactoring should be in order, to allow both `CreateVolume()` and `ValidateVolumeCapabilities()` to use the same validate function.

Reply:
+1

And an unit test would be great.

Reply:
I don't think we can use volCapabilities for this. volCapabilities are the published, *generic* capabilities of the driver. What we need to check here are the *specific* capabilities of the volume type.

We fetch the id of the volume type below. I think we need to pull the extra_specs of the volume type and specifically check for multiattach support.

Reply:
I've written up what I believe should be the correct test here: https://github.com/kubernetes/cloud-provider-openstack/pull/1368#issuecomment-818831513